components/properties/src/maps.rs

```diff
@@ -289,16 +298,14 @@ impl<'a> CodePointMapDataBorrowed<'a, crate::GeneralCategory> {
     /// assert_eq!(ranges.next().unwrap(), 'µ' as u32..='µ' as u32);
     /// assert_eq!(ranges.next().unwrap(), 'º' as u32..='º' as u32);
     /// assert_eq!(ranges.next().unwrap(), 'À' as u32..='Ö' as u32);
-    /// assert_eq!(ranges.next().unwrap(), 'Ø' as u32..='Þ' as u32);
+    /// assert_eq!(ranges.next().unwrap(), 'Ø' as u32..='ö' as u32);
     /// ```
     pub fn iter_ranges_for_group(
         self,
         val: crate::GeneralCategoryGroup,
     ) -> impl Iterator<Item = RangeInclusive<u32>> + 'a {
         self.map
-            .iter_ranges()
-            .filter(move |r| val.contains(r.value))
-            .map(|r| r.range)
+            .iter_ranges_for_predicate(move |r| val.contains(r.value))
```

Comment:
thought: would it make sense to instead of adding `iter_ranges_for_predicate` to make the `filter` method on whatever `iter_ranges` returns behave like this?

Reply:
It's possible to do, requries a manual iterator, and is not quite following the docs of `filter()` since it's no longer a true filtering operation. I don't think we should do that.

Reply:
Could also be a non-trait method that takes whatever type `val` is.

Reply:
As in?

Reply:
https://github.com/unicode-org/icu4x/issues/1532 has a slightly different proposal: a way to _map_ the value and have it coalesce ranges, in this case we could map to a bool and then filter.

I think that's more generic and could add that here but I don't really think it's necessary.